File: processor/src/test/resources/approvals/beanvalidation/IsbnValueExtractor.java
package test;

import jakarta.validation.valueextraction.ExtractedValue;
import jakarta.validation.valueextraction.UnwrapByDefault;
import jakarta.validation.valueextraction.ValueExtractor;
import scenarios.java.Isbn;

import javax.annotation.processing.Generated;

@Generated("com.qualityminds.lazyval.processor.internal.codegen.beanvalidation.BeanValidationGenerator")
@UnwrapByDefault
public class IsbnValueExtractor implements ValueExtractor<@ExtractedValue(type = String.class) Isbn> {
  @Override
  public void extractValues(Isbn originalValue, ValueExtractor.ValueReceiver receiver) {
    if (originalValue == null) {
      receiver.value(null, null);
      return;
    }
    receiver.value(null, originalValue.getValue());
  }
}

File: ksp/src/test/resources/approvals/beanvalidation/IsbnValueExtractor.kt
package test.domain

import jakarta.`annotation`.Generated
import jakarta.validation.valueextraction.ValueExtractor
import jakarta.validation.valueextraction.UnwrapByDefault
import scenarios.kotlin.Isbn

@Generated("com.qualityminds.lazyval.ksp.internal.codegen.beanvalidation.BeanValidationGenerator")
@UnwrapByDefault
public class IsbnValueExtractor : IsbnValueExtractorBase() {
  override fun extractValues(originalValue: Isbn?, `receiver`: ValueExtractor.ValueReceiver) {
    if (originalValue == null) {
      receiver.value(null, null)
      return
    }
    receiver.value(null, originalValue.value)
  }
}
